lookup.go

```diff
@@ -76,7 +75,7 @@ func (dht *IpfsDHT) GetClosestPeers(ctx context.Context, key string) (<-chan pee
 	e := logger.EventBegin(ctx, "getClosestPeers", loggableKey(key))
```

Comment:
I agree that this is tricky, especially for `Provide` and `PutValue` since they imply replication semantics. I'm less concerned about `GetClosestPeers` since it has no inherent semantics and could just get a documentation change (also given it's lack of semantics I doubt it's in real use see [here](https://grep.app/search?q=.GetClosestPeers&case=true&filter[lang][0]=Go))

WDYT @petar?

Reply:
My understanding of the decision here was that even though the internals have changed the guarantees are similar and the nuances are both not something the consumer of the DHT should care about and are a bit much to just throw in as a documentation comment. If we want we can add comments to GetClosestPeers since it's really an implementation specific function (i.e. the contract is, and probably should be, tightly coupled to the implementation). 

We are planning on rewriting the DHT spec once the new protocol version is released (or at least in the RC phase) and IMO detailing the specifics there or in a sister document would be excellent.